src/backend/src/services/reimbursement-requests.services.ts

```diff
@@ -880,4 +880,30 @@ export default class ReimbursementRequestService {
 
     return vendorTransformer(vendor);
   }
+
+  /**
+   * Deletes the vendor
+   *
+   * @param vendorId the requested vendor to be deleted
+   * @param submitter the user deleting the vendor
+   * @returns the 'deleted' vendor
+   */
+  static async deleteVendor(vendorId: string, submitter: User) {
+    await isUserAdminOrOnFinance(submitter);
+
+    const vendor = await prisma.vendor.findUnique({
+      where: { vendorId }
+    });
+
+    if (!vendor) throw new NotFoundException('Vendor', vendorId);
+
+    if (vendor.dateDeleted) throw new DeletedException('Vendor', vendorId);
```

Comment:
can you add a check like this to editVendor as well, and rename it to 'editVendor' from 'editVendors'